actor-profile.js

```diff
@@ -0,0 +1,96 @@
+import { db } from "./dbInstance.js";
+import { fetchActorInfo } from "./post.js";
+
+class ActorProfile extends HTMLElement {
+  static get observedAttributes() {
+    return ["url"];
+  }
+
+  constructor() {
+    super();
+    this.url = "";
+  }
+
+  connectedCallback() {
+    this.url = this.getAttribute("url");
+    this.fetchAndRenderActorProfile(this.url);
+  }
+
+  async fetchAndRenderActorProfile(url) {
+    const actorInfo = await fetchActorInfo(url);
+    console.log(actorInfo);
+    if (actorInfo) {
+      this.renderActorProfile(actorInfo);
+      this.updateFollowButtonState();
+      // Update distributed-outbox URL based on fetched actorInfo
+      const distributedOutbox = document.querySelector("distributed-outbox");
```

Comment:
The outbox should be a child of this component and we should use `this.querySelector`